src/Analysis/Ast/Impl/Analyzer/Evaluation/ExpressionEval.Collections.cs

```diff
@@ -101,40 +102,47 @@ internal sealed partial class ExpressionEval {
             return UnknownType;
         }
 
-        public IMember GetValueFromComprehension(Comprehension node) {
-            var oldVariables = CurrentScope.Variables.OfType<Variable>().ToDictionary(k => k.Name, v => v);
-            try {
-                ProcessComprehension(node);
-
-                // TODO: Evaluate comprehensions to produce exact contents, if possible.
-                switch (node) {
-                    case ListComprehension lc:
-                        var v1 = GetValueFromExpression(lc.Item) ?? UnknownType;
-                        return PythonCollectionType.CreateList(Module, new[] { v1 });
-                    case SetComprehension sc:
-                        var v2 = GetValueFromExpression(sc.Item) ?? UnknownType;
-                        return PythonCollectionType.CreateSet(Module, new[] { v2 });
-                    case DictionaryComprehension dc:
-                        var k = GetValueFromExpression(dc.Key) ?? UnknownType;
-                        var v = GetValueFromExpression(dc.Value) ?? UnknownType;
-                        return new PythonDictionary(new PythonDictionaryType(Interpreter.ModuleResolution.BuiltinsModule), new Dictionary<IMember, IMember> { { k, v } });
+        public IMember GetValueFromComprehensionInScope(Comprehension node, bool is3X = true) {
```

Comment:
Same as above. Could be the same name, and read is3X internaly, unless I've missed a usage.